src/CanvasRenderingContext2d.cc

```diff
@@ -1884,6 +1884,11 @@ NAN_METHOD(Context2d::ClosePath) {
  */
 
 NAN_METHOD(Context2d::Rotate) {
+  double angle = info[0]->NumberValue();
+
+  if (isnan(angle) || isinf(angle))
+    return;
+
   Context2d *context = Nan::ObjectWrap::Unwrap<Context2d>(info.This());
   cairo_rotate(context->context()
     , info[0]->IsNumber() ? info[0]->NumberValue() : 0);
```

Comment:
This line should probably be changed as well. 

btw, in the browser, if you set it to NaN, does it set it to 0 or just don't change it?

Reply:
@LinusU 

> This line should probably be changed as well.

Sorry, which line?

> In the browser, if you set it to NaN, does it set it to 0 or just don't change it?

Well, the easiest way to find it out is to check manually. All modern browsers (which correctly follow specification for HTML5 canvas) simply ignore NaNs as rotation angles. They don't set the angle to 0, just return from the function without taking any actions. I posted the link to the part from the specification which defines behavior in case of NaN as the argument.

Reply:
Sorry, missed the specification link, that's perfect!

> Sorry, which line?

```c++
cairo_rotate(context->context()
     , info[0]->IsNumber() ? info[0]->NumberValue() : 0);
```

could be updated to just:

```c++
cairo_rotate(context->context(), angle);
```

Also, it would be nice to double check what `info[0]->NumberValue()` returns if `info[0]-IsNumber() == false`. I would guess that it returns `NaN`, but if you could add a test case that tries setting the rotation to a string, boolean, array, etc. and see that it's being properly ignored, that would be good 👍

Reply:
@LinusU 
Right, that check was completely redundant, I've just removed it.

> I would guess that it returns `NaN`

What actually happens is that javascript engine calls  `Symbol.toPrimitive` of it's prototype and converts the returned value to a number. For strings it would try to parse it as a double, for arrays it will stringify the array and again convert it to a double, etc.

> If you could add a test case that tries setting the rotation to a string, boolean, array, etc. and see that it's being properly ignored, that would be good

Ok, I'll add a test case, but not all non-number values should be ignored, as I explained, only if the return value of `Symbol.toPrimitive` is NaN or infinite.

Reply:
Ah, cool, yeah it seems like the browser accepts e.g. `'20'`. Good to know :)

Perfect then! Thank you for your work on this!